[CORE-14743] dt/dl: Crank the timeout in DatalakeCustomPartitioningTest.test_many_partitions

oleiman opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

Crank up the progress timeout in datalake test for pathological custom partitioning config.

CDT looks good: https://buildkite.com/redpanda/redpanda/builds/77206#019ae090-395f-4f57-92cb-64dd65574d4d

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

oleiman committed:
dt/util: Improve logging in wait_until_with_progress_check

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman committed:
dt/dl: Crank progress timeout in custom partitioning many partitions test

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman commented:
/ci-repeat 1
release
skip-units
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_many_partitions

oleiman commented:
/cdt
dt-repeat=10
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_many_partitions

oleiman commented:
/cdt
dt-repeat=10
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_many_partitions

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--19af8d91-06a8-4eab-9b2c-614fef2ed03d-->
<!--TEST RESULTS-->
<!--3a3f035a-f4a6-44da-8437-2df13a670e4f-->
<!--TEST RESULTS-->
<!--2b62c686-3503-41c5-8933-ea57c14a5096-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77211">build#77211</a></summary>


| test_class                    | test_method       | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    | test_history                                                                                                                                                  |
|:------------------------------|:------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_with_restart | null             | integration | https://buildkite.com/redpanda/redpanda/builds/77211#019ae0d7-b005-45eb-ada2-78a2da884757 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_with_restart |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77237">build#77237</a></summary>


| test_class                 | test_method                     | test_arguments                          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     | test_history                                                                                                                                                             |
|:---------------------------|:--------------------------------|:----------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "LEADERSHIP_TRANSFER"} | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae314-4610-45ec-84d8-7dd04af366be | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "LEADERSHIP_TRANSFER"} | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae31a-6394-4a7f-aee2-79a068f82a56 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "NONE"}                | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae314-4612-4f8f-86f7-919488955385 | FLAKY         | 1/21     | upstream reliability is '100.0'. current run reliability is '4.761904761904762'. drift is 95.2381 and the allowed drift is set to 50. The test should FAIL | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "NONE"}                | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae31a-6396-4431-b835-a3908b7b52f2 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "RANDOM"}              | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae314-4613-43bc-9ce4-33164476fb48 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
| DirectConsumerVerifierTest | test_basic_consuming_from_topic | {"failure_mode": "RANDOM"}              | integration | https://buildkite.com/redpanda/redpanda/builds/77237#019ae31a-6398-4860-83a7-6a07b0738d1a | FAIL          | 0/21     | The test has failed across all retries                                                                                                                     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DirectConsumerVerifierTest&test_method=test_basic_consuming_from_topic |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/77268">build#77268</a></summary>


| test_class                 | test_method                              | test_arguments          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                      |
|:---------------------------|:-----------------------------------------|:------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ScalingUpTest              | test_fast_node_addition                  | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/77268#019ae539-8c1e-427a-9051-9b850bf4532b | FLAKY         | 19/21    | upstream reliability is '97.01986754966887'. current run reliability is '90.47619047619048'. drift is 6.54368 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ScalingUpTest&test_method=test_fast_node_addition                               |
| TxAtomicProduceConsumeTest | test_basic_tx_consumer_transform_produce | {"with_failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/77268#019ae540-399a-4d60-951a-6e648e0beed9 | FLAKY         | 20/21    | upstream reliability is '98.29683698296837'. current run reliability is '95.23809523809523'. drift is 3.05874 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TxAtomicProduceConsumeTest&test_method=test_basic_tx_consumer_transform_produce |
</details>
<!--/body-->

nvartolomei commented on tests/rptest/util.py:
```diff
@@ -162,7 +162,8 @@ def wait_until_with_progress_check(
     val = check()
-    while timeout_sec > 0:
+    elapsed_sec = 0
+    while elapsed_sec < timeout_sec:
```
This method seems buggy. If timeout is 30s and progress check is 20s it can run as follows:

Condition v == 100

Initial check: v = 1
Second check at 20s: condition fails, progress check v = 2 | ok
Third check is at 40s (questionable): condition fails, progress check succeeds v = 3 | ok

Now elapsed sec is greater than timeout, the function exits without an exception although condition (v == 100) is not satisfied.

oleiman replied:
ugh good eye

oleiman replied:
@nvartolomei should be good to go now, but feel free to wait until it clears CI

oleiman committed:
dt/util: Fix bug in wait_until_with_progress_check

Previously the check could pass in situations where the value is making
progress but the condition has not yet been reached.

Now we always check the condition on the clean return path.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/77237-->
<!--c6cf1d02-2c93-4115-b417-88949c0d65ca-->
<!--c5191d0e-187e-4b76-897b-fa4e195bdde4-->
<!--9e85eecf-cd96-4b34-9231-10cef2c63eb9-->
<!--cb71d2d2-56fd-4209-8e3b-409fa2660d1a-->
<!--8a4f6b4c-c131-4be1-ac74-a42a93c600dd-->
<!--b4d31967-36ad-4802-9d98-93b32454719c-->
<!--title-->
## Retry command for [Build#77237](https://buildkite.com/redpanda/redpanda/builds/77237)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
skip-redpanda-build
skip-units
skip-rebase
tests/rptest/direct_consumer_tests/direct_consumer_test.py::DirectConsumerVerifierTest.test_basic_consuming_from_topic@{"failure_mode":"NONE"}
tests/rptest/direct_consumer_tests/direct_consumer_test.py::DirectConsumerVerifierTest.test_basic_consuming_from_topic@{"failure_mode":"RANDOM"}
tests/rptest/direct_consumer_tests/direct_consumer_test.py::DirectConsumerVerifierTest.test_basic_consuming_from_topic@{"failure_mode":"LEADERSHIP_TRANSFER"}
```
<!--/body-->

nvartolomei commented:
I assume failures are related @oleiman thus skipping review for now. Lmk when ready.

oleiman commented:
@nvartolomei fire when ready 🙏

vbotbuildovich commented:
/backport v25.3.x
